playkit/src/main/java/com/kaltura/playkit/player/PKSubtitlePosition.java

```diff
@@ -0,0 +1,182 @@
+package com.kaltura.playkit.player;
+
+import android.text.Layout;
+
+import com.kaltura.android.exoplayer2.text.Cue;
+import com.kaltura.playkit.utils.Consts;
+
+public class PKSubtitlePosition {
+
+    // If `overrideInlineCueConfig` is set to true, player will ignore the existing values coming in Cue Settings
+    // and will always use the given Cue Settings
+    public PKSubtitlePosition(boolean overrideInlineCueConfig) {
+        this.overrideInlineCueConfig = overrideInlineCueConfig;
+    }
+
+    // For Left to Right Language texts(LTR), Allow the subtitle view to move left (ALIGN_NORMAL), middle (ALIGN_CENTER) and right (ALIGN_OPPOSITE)
+    // For Right to Left Language texts(RTL), Allow the subtitle view to move Right (ALIGN_NORMAL), middle (ALIGN_CENTER) and left (ALIGN_OPPOSITE)
+    private Layout.Alignment subtitleHorizontalPosition = Layout.Alignment.ALIGN_CENTER;
+
+    // Allow the subtitle view to move vertically (100% = 100 OR 10% = 10)
+    // For vertical positions, percentage starts from Top(100) to Bottom(10)
+    private float verticalPositionPercentage = Cue.DIMEN_UNSET;
+
+    // Allow the subtitle view to move horizontally but with in the left-middle and middle-right viewport
+    // For horizontal(Left viewport) positions, percentage starts from center(10 or 10%) to Left(100 or 100%)
+    // For horizontal(Right viewport) positions, percentage starts from center(10 or 10%) to Right(100 or 100%)
+    private float horizontalPositionPercentage = Cue.DIMEN_UNSET;
+
+    // The type of line, Fraction, Number, Unset. We are accepting values for Line in fraction so
+    // keeping it LINE_TYPE_FRACTION. It is not exposed.
+    private int lineType = Cue.LINE_TYPE_FRACTION;
+
+    // Lower limit either for vertical (top to bottom) for horizontal (center to left/right)
+    private int positionLowerLimit = 10;
+
+    // Override the current subtitle Positioning with the In-stream subtitle text track configuration
+    private boolean overrideInlineCueConfig;
+
+    public Layout.Alignment getSubtitleHorizontalPosition() {
+        return subtitleHorizontalPosition;
+    }
+
+    public float getVerticalPositionPercentage() {
+        return verticalPositionPercentage;
+    }
+
+    public float getHorizontalPositionPercentage() {
+        return horizontalPositionPercentage;
+    }
+
+    public boolean isOverrideInlineCueConfig() {
+        return overrideInlineCueConfig;
+    }
+
+    public int getLineType() {
+        return lineType;
+    }
```

Comment:
I don't think we need to expose those getters.

Reply:
We made it protected.